### PR TITLE
fix: various custom operator conformance fixes

### LINKED ISSUE #1515: Update flagd-testbed to v3.6.2 and fix in-process evaluation edge cases
## Summary

Update the flagd-testbed dependency to [v3.6.2](https://github.com/open-feature/flagd-testbed/releases/tag/v3.6.2) and fix the in-process evaluation to pass all new edge-case tests.

Testbed v3.6.2 adds coverage for the edge cases identified in:
- open-feature/flagd#1873 (sem_ver parsing inconsistencies)
- open-feature/flagd#1874 (custom operator error return values)
- open-feature/flagd#1875 ($evaluators/$ref resolution gaps)

### sem_ver (open-feature/flagd#1873)

1. **sem_ver returns `false` on parse failure instead of `null`**; should return `null` to trigger default variant fallback
2. **sem_ver returns `false` on unknown operator instead of `null`**; same issue as above

### Custom operator error returns (open-feature/flagd#1874)

3. **`starts_with` returns `false` on error instead of `null`**; should return `null`
4. **`ends_with` returns `false` on error instead of `null`**; should return `null`

### $evaluators/$ref resolution (open-feature/flagd#1875)

5. **No diagnostic feedback on unresolved `$ref`s at parse time**; currently throws during `engine.build()` with misleading error message ("Method '$ref' was not found in the Logic Engine") rather than identifying the missing evaluator name

Parent: open-feature/flagd#1904

### PR DESCRIPTION
**Pull request by toddbaert**
* support v/V prefix in semver
* support version "metadata" in semver
* fix whitespace handling in $refs
* return null on errors

:warning: wait until flagd/RPC is updated to merge.

Fixes: https://github.com/open-feature/js-sdk-contrib/issues/1515

---

Related:
* https://github.com/open-feature/flagd/pull/1950
* https://github.com/open-feature/java-sdk-contrib/pull/1778
* https://github.com/open-feature/dotnet-sdk-contrib/pull/635
* https://github.com/open-feature/python-sdk-contrib/pull/386
* https://github.com/open-feature/go-sdk-contrib/pull/874

### CODE REVIEW

File: libs/shared/flagd-core/src/lib/parser.ts [hunk 1]
```diff
@@ -98,7 +98,7 @@ function transform(flagCfg: string): string {
 
   for (const key in evaluators) {
     const replacer = JSON.stringify(evaluators[key]).replaceAll(bracketReplacer, '');
-    const refRegex = new RegExp('"\\$ref":(\\s)*"' + key + '"', 'g');
+    const refRegex = new RegExp('"\\$ref"(\\s)*:(\\s)*"' + key + '"', 'g');
 
     transformed = transformed.replaceAll(refRegex, replacer);
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply libs/shared/flagd-core/src/lib/parser.ts]: Fixed with a JSON.stringify/parse round trip, as in other impls.

**Review by leakonvalinka — APPROVED** [review]

**Review by marcozabel — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]
